homeassistant/components/whirlpool/__init__.py

```diff
@@ -0,0 +1,61 @@
+"""The Whirlpool Sixth Sense integration."""
+import asyncio
+import logging
+
+import aiohttp
+from whirlpool.auth import Auth
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import AUTH_INSTANCE_KEY, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["climate"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Whirlpool Sixth Sense component."""
+    hass.data.setdefault(DOMAIN, {})
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Whirlpool Sixth Sense from a config entry."""
+
+    auth = Auth(entry.data["username"], entry.data["password"])
+    try:
+        await auth.load_auth_file()
+    except aiohttp.ClientConnectionError:
+        _LOGGER.error("Connection error")
+        return False
+
+    if not auth.is_access_token_valid():
```

Comment:
Is this doing I/O ?

Reply:
no, it just checks if it has an access token and its expire date, both of which are already in memory after auth is done.